system-addon/lib/AutoMigrateFeed.jsm

```diff
@@ -0,0 +1,50 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+"use strict";
+
+const {utils: Cu} = Components;
+Cu.import("resource://gre/modules/XPCOMUtils.jsm");
+const {actionTypes: at, actionCreators: ac} = Cu.import("resource://activity-stream/common/Actions.jsm", {});
+
+XPCOMUtils.defineLazyModuleGetter(this, "AutoMigrate",
+  "resource:///modules/AutoMigrate.jsm");
+XPCOMUtils.defineLazyModuleGetter(this, "Services",
+  "resource://gre/modules/Services.jsm");
+
+this.AutoMigrateFeed = class AutoMigrateFeed {
+  onAction(action) {
+    switch (action.type) {
+      case at.AUTOMIGRATE_MIGRATE_DONE:
+        AutoMigrate.keepAutoMigration();
+        this.store.dispatch(ac.BroadcastToContent({type: at.AUTOMIGRATE_HIDE}));
+        break;
+      case at.AUTOMIGRATE_UNDO_MIGRATION:
+        AutoMigrate.undoAutoMigration(action._target.browser.ownerGlobal);
+        this.store.dispatch(ac.BroadcastToContent({type: at.AUTOMIGRATE_IS_REVERTED}));
+        break;
+      case at.NEW_TAB_VISIBLE:
+        AutoMigrate.shouldShowMigratePrompt(action._target.browser).then(prompt => {
+          if (prompt) {
+            let browserName = Services.prefs.getStringPref("browser.migrate.automigrate.browser", "");
```

Comment:
Why is there this pref logic when home/newtab usage doesn't check? https://dxr.mozilla.org/mozilla-central/search?q=%2Bref%3A%22AutoMigrate%23shouldShowMigratePrompt%22

The logic seems a bit odd in having Activity Stream seemingly randomly check a pref to do something different and somewhat overriding the just called `shouldShowMigratePrompt`. Is it possible to have that `should` method indicate what type of prompt should be shown instead of that a prompt should be shown?